python_modules/libraries/dagster-k8s/dagster_k8s/component.py

```diff
@@ -0,0 +1,62 @@
+from collections.abc import Mapping, Sequence
+from dataclasses import dataclass
+from functools import cached_property
+from typing import Any, Optional, Union
+
+from dagster import AssetExecutionContext, AssetsDefinition, Definitions, multi_asset
+from dagster.components import Component, ComponentLoadContext, Resolvable, ResolvedAssetSpec
+
+from dagster_k8s.pipes import PipesK8sClient, build_pod_body
+
+
+@dataclass
+class PipesK8sSpec(Resolvable):
+    name: str
+    assets: Sequence[ResolvedAssetSpec]
+    image: Optional[str] = None
+    command: Optional[Union[str, Sequence[str]]] = None
+    namespace: Optional[str] = None
+    env: Optional[Mapping[str, str]] = None
+    base_pod_meta: Optional[Mapping[str, Any]] = None
+    base_pod_spec: Optional[Mapping[str, Any]] = None
+
+    def __post_init__(self):
+        # validate that we can build a pod for the given args
+        # i.e. image or base_pod_spec.image
+        build_pod_body(
+            pod_name=self.name,
+            image=self.image,
+            command=self.command,
+            env_vars=self.env or {},
+            base_pod_meta=self.base_pod_meta,
+            base_pod_spec=self.base_pod_spec,
+        )
+
+
+@dataclass
+class PipesK8sCollectionComponent(Component, Resolvable):
```

Comment:
I think we should decide on name from first principles here and change the `dagster.components.PipesSubprocessScriptCollectionComponent` match. I think the existing convention is quite ugly.

Reply:
`K8sPodsComponent` ?

Reply:
`PipesSubprocessScriptCollectionComponent` --> `PythonScriptsComponent`?

Reply:
The other alt here is some sort of scheme where there is a an individual `K8sPodComponent` and `PythonScriptComponent` and a generic composability thing on top of it, which would feel great I think. I know @OwenKephart had some thoughts in that direction.